pkg/tls/mutal.go

```diff
@@ -0,0 +1,218 @@
+	if err != nil {
+		return fmt.Errorf("unable to generate RSA key: %v", err)
+	}
+
+	certificateSigningInfo := &openssl.CertificateInfo{
+		Serial:       i64,
+		Issued:       0,
+		Expires:      0,
+		Country:      g.Country,
+		Organization: g.Organization,
+		CommonName:   g.CommonName,
+	}
+	caCert, err := openssl.NewCertificate(certificateSigningInfo, caKey)
+	if err != nil {
+		return fmt.Errorf("unable to generate new signing certificate: %v", err)
+	}
+	name := &openssl.Name{}
+	name.AddTextEntries(g.SubjectFields)
+	caCert.SetSubjectName(name)
+	caCert.SetVersion(openssl.X509_V3)
+	err = caCert.Sign(caKey, openssl.EVP_SHA256)
+	if err != nil {
+		return fmt.Errorf("unable to sign caCert: %v", err)
+	}
+
+	//$ openssl genrsa -passout pass:1234 -des3 -out server.key 4096
+	serverKey, err := openssl.GenerateRSAKey(g.RSABytes)
+	if err != nil {
+		return fmt.Errorf("unable to generate server RSA key: %v", err)
+	}
+
+	//$ openssl req -passin pass:1234 -new -key server.key -out server.csr -subj  "/C=SP/ST=Italy/L=Ornavasso/O=Test/OU=Server/CN=localhost"
+	serverCertificateSigningInfo := &openssl.CertificateInfo{
+		Serial:       i64,
+		Issued:       0,
+		Expires:      0,
+		Country:      g.Country,
+		Organization: g.Organization,
+		CommonName:   g.CommonName,
+	}
+	serverCSR, err := openssl.NewCertificate(serverCertificateSigningInfo, serverKey)
+	if err != nil {
+		return fmt.Errorf("unable to generate new signing certificate: %v", err)
+	}
+	serverName := &openssl.Name{}
+	serverName.AddTextEntries(g.SubjectFields)
+	serverCSR.SetSubjectName(name)
+	serverCSR.SetVersion(openssl.X509_V3)
+	err = serverCSR.Sign(serverKey, openssl.EVP_SHA256)
+	if err != nil {
+		return fmt.Errorf("unable to sign serverCSR: %v", err)
+	}
+
+	//$ openssl x509 -req -passin pass:1234 -days 365 -in server.csr -CA ca.crt -CAkey ca.key -set_serial 01 -out server.crt
+
+	serverCASigningInfo := &openssl.CertificateInfo{
+		Serial:       i64,
+		Issued:       0,
+		Expires:      0,
+		Country:      g.Country,
+		Organization: g.Organization,
+		CommonName:   g.CommonName,
+	}
+	serverCert, err := openssl.NewCertificate(serverCASigningInfo, caKey)
+	if err != nil {
+		return fmt.Errorf("unable to create new server cert: %v", err)
+	}
+	serverCert.SetIssuer(caCert)
+	err = caCert.Sign(caKey, openssl.EVP_SHA256)
+	if err != nil {
+		return fmt.Errorf("unable to sign caCert: %v", err)
+	}
+	err = serverCert.Sign(caKey, openssl.EVP_SHA256)
+	if err != nil {
+		return fmt.Errorf("unable to sign serverCert: %v", err)
+	}
+
+	// Cache TLS Cert material
+	g.CACert = caCert
+	g.ServerCert = serverCert
+	g.ServerCSR = serverCSR
+	g.CAKey = caKey
+	g.ServerKey = serverKey
+
+	return nil
+}
+
+const (
+	DefaultServerKey  = "server.key"
+	DefaultCAKey      = "ca.key"
+	DefaultServerCSR  = "server.csr"
+	DefaultCACert     = "ca.crt"
+	DefaultServerCert = "server.crt"
+)
+
+func (g *GRPCTLSGenerator) FlushToDisk(path string) error {
```

Comment:
There are 5 small "systems" here...

Perhaps we pull these systems into a function with arguments for readability?

Reply:
Also as per usual, I hate this method name

`CommitLocal` 
` WriteFiles`
`WriteToPath`
`WriteFilesToPath`
`CommitFilesToPath`